Versions changes

nirarg opened the pull request:
**What this PR does / why we need it**:
1. Use capi version v1beta1 (changed from v1alpha4)
2. Use Go version 1.17 (changed from 1.16)

rmohr commented on config/default/manager_image_patch.yaml:
```diff
@@ -8,5 +8,5 @@ spec:
       containers:
       # Change the value of image field below to your controller image URL
-      - image: gcr.io/k8s-staging-cluster-api/capk-manager:dev
+      - image: localhost:49293/capk-manager-amd64:dev
```
This change seems to be not intentional.

nirarg replied:
You right, removed this change

rmohr commented on go.mod:
```diff
@@ -1,30 +1,81 @@
+	kubevirt.io/controller-lifecycle-operator-sdk v0.2.1 // indirect
+	sigs.k8s.io/json v0.0.0-20211020170558-c049b76a60c6 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.1.2 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
```
There are a lot of replaces. What exactly is the reason for this? Is it the v1beta1 api from capi?

nirarg replied:
The section you pointed on is "require" section
This is auto created after the update to go 1.17
There are 2 `require` sections which are auto generated 
There is one `replace` section which include only 3 lines

rmohr replied:
> The section you pointed on is "require" section

Indeed :+1:

rmohr reviewed (changes requested):
Look pretty good overall. A few comments inline.

nirarg commented:
/cc @cchengleo

davidvossel reviewed (approved):
/lgtm

I don't have lgtm permissions yet, but take that as an offer of my support :D 

The code looks good to me and func tests passed for me locally. We'll need an OWNER to issue the `/ok-to-test` command in order for CI to execute the e2e tests

agradouski commented:
/ok-to-test

rmohr commented:
@nirarg  I need to fix something in the integration test. I have to bump it to 1.17 too: 

```
../../../../pkg/mod/k8s.io/client-go@v0.23.0-alpha.4/plugin/pkg/client/auth/exec/metrics.go:21:2: package io/fs is not in GOROOT (/root/.gimme/versions/go1.15.8.linux.amd64/src/io/fs)
```

nirarg commented:
> @nirarg I need to fix something in the integration test. I have to bump it to 1.17 too:

I rebased the PR on this fix and changed Golang version in the Dockerfile

nirarg commented:
/hold
Merge decreasing to v1alpha1 version in different PR:
https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/pull/54

nirarg commented:
/hold cancel
@cchengleo I rebased this PR after #54 was merged
Can you please test this too?

cchengleo reviewed (approved):
I also verified this PR in our CI and passed. Thanks for the great job!

cchengleo commented:
/lgtm
